src/OrbitGl/App.cpp

```diff
@@ -1147,166 +1159,145 @@ ErrorMessageOr<std::filesystem::path> OrbitApp::FindSymbolsLocally(
+}
+
+orbit_base::Future<ErrorMessageOr<void>> OrbitApp::LoadSymbols(
+    const std::filesystem::path& symbols_path, const std::string& module_file_path) {
+  auto scoped_status = CreateScopedStatus(absl::StrFormat(
+      R"(Loading symbols for "%s" from file "%s"...)", module_file_path, symbols_path.string()));
+
+  auto future = thread_pool_->Schedule(
+      [symbols_path]() { return SymbolHelper::LoadSymbolsFromFile(symbols_path); });
+
+  return future.Then(
+      main_thread_executor_,
+      [this, module_file_path, scoped_status = std::move(scoped_status)](
+          const ErrorMessageOr<orbit_grpc_protos::ModuleSymbols>& module_symbols) mutable
+      -> ErrorMessageOr<void> {
+        if (!module_symbols) return module_symbols.error();
+
+        ModuleData* module_data = GetMutableModuleByPath(module_file_path);
+        module_data->AddSymbols(module_symbols.value());
+
+        std::string message =
+            absl::StrFormat(R"(Successfully loaded %d symbols for "%s")",
+                            module_symbols.value().symbol_infos_size(), module_data->file_path());
+        scoped_status.UpdateMessage(message);
+        LOG("%s", message);
+
+        const ProcessData* selected_process = GetTargetProcess();
+        if (selected_process != nullptr &&
+            selected_process->IsModuleLoaded(module_data->file_path())) {
+          functions_data_view_->AddFunctions(module_data->GetFunctions());
+          LOG("Added loaded function symbols for module \"%s\" to the functions tab",
+              module_data->file_path());
+        }
 
-          UpdateAfterSymbolLoading();
-          FireRefreshCallbacks();
-        });
-  });
+        UpdateAfterSymbolLoading();
+        FireRefreshCallbacks();
+        return outcome::success();
+      });
 }
 
-ErrorMessageOr<void> OrbitApp::GetFunctionInfosFromHashes(
-    const ModuleData* module, const std::vector<uint64_t>& function_hashes,
-    std::vector<const FunctionInfo*>* function_infos) {
-  CHECK(module != nullptr);
-  const ProcessData* process = GetTargetProcess();
-  if (process == nullptr) {
-    return ErrorMessage(absl::StrFormat(
-        "Unable to get function infos for module \"%s\", because no process is selected",
-        module->file_path()));
+OrbitApp::LoadPresetModuleResult OrbitApp::LoadPresetModule(
+    const std::string& module_path, const orbit_client_protos::PresetModule& preset_module) {
+  ModuleData* module_data = module_manager_->GetMutableModuleByPath(module_path);
+  if (module_data == nullptr) {
+    return LoadPresetModuleResult::kModuleNotFound;
   }
-  if (!process->IsModuleLoaded(module->file_path())) {
-    return ErrorMessage(absl::StrFormat(
-        R"(Unable to get function infos for module "%s", because the module is not loaded by process "%s")",
-        module->file_path(), process->name()));
+
+  if (!module_data->is_loaded()) {
+    const auto result = main_thread_executor_->WaitFor(LoadModule(module_data));
```

Comment:
`WaitFor()` means that this function potentially takes a long time right?. Its not blocking, but "waiting". This is fine, because this function is only used inside a future (in `LoadPreset`), but maybe we should start indicating that in the function name. Maybe `LoadPresetModuleWaiting()`. 
Or maybe this function should have return type `Future<OrbitApp::LoadPresetModuleResult>`? (Did not fully think this through)